test_xmlsec.py

```diff
@@ -78,3 +112,26 @@ def test_sign_and_verify_with_sha1():
     private_key = "/tmp/https/tls.key"
     assert sign_xml_template(outfile, template, private_key) == 0
     assert verify_xml_template(outfile) == 0
+
+def test_sign_and_verify_with_ecdsa_sha384():
+    """Test you can sign and verify xml with ecdsa sha386
+
+    :id: ca201596-4663-49d3-9c4a-615a9a70aa97
+    :steps:
+        1. Generate ECDSA private key
+        2. Derive public key from private key
+        3. Sign xml template using ecdsa sha486
+        4. Verify signed file using public key
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+    """
+    template = "xmlsec_data/book-template-ecdsa-sha384.xml"
+    outfile = get_outfile(template)
+    private_key = "/tmp/https/xmlsec_ecdsa.key"
+    public_key = "/tmp/https/xmlsec_ecdsa.pub"
+    gen_key_ecdsa(private_key)
```

Comment:
Generally looks great. Just minor comment -- `gen_key_ecdsa` and `gen_pubkey_ecdsa` returns an error code. If we implemented returning code, we should check it.

`assert gen_key_ecdsa(private_key) == 0` and `assert gen_pubkey_ecdsa(private_key) == 0` would be enough.

Reply:
Updated.